src/components/InternalLinkButton/InternalLinkButton.tsx

```diff
@@ -0,0 +1,31 @@
+import Link, { LinkProps } from 'next/link';
+import { Button, ButtonProps } from '@aws-amplify/ui-react';
+
+interface InternalLinkButtonProps extends LinkProps {
+  variation?: ButtonProps['variation'];
+  size?: ButtonProps['size'];
+  children?: React.ReactNode;
+  className?: string;
+}
+
+export const InternalLinkButton = ({
+  variation,
+  size,
+  href,
+  children,
+  className
+}: InternalLinkButtonProps) => {
+  return (
+    <Link href={href} legacyBehavior={true} passHref={true}>
+      <Button
+        variation={variation}
+        gap="small"
+        size={size}
+        as="a"
```

Comment:
Do we still need to use `as="a"` here since we're already wrapping the button with the `next/link`?

Reply:
We still need the `as="a"` otherwise it is output as a `<button>` element with an href.

Reply:
Oh I see, so you just want it to be an `<a>` tag only instead of it being `<a><button></button></a>` ?